### ISSUE TITLE
`no-props-default-factory-undefined`

### ISSUE THREAD

**Issue by fisker** [vuejs/eslint-plugin-vue#2353]
**Please describe what the rule should do:**

Disallow `() => {}` as the default of props.

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
defineProps({
  foo: {
   default: () => {},
  },
});
```

probably mean to use the following, but forgot parentheses

```vue
defineProps({
  foo: {
   default: () => ({}),
  },
});
```

**Additional context**

In most cases, it's just a mistake to return an empty object. If mean to default to `undefined`,

```vue
defineProps({foo: {default: undefined}});
```

OR

```vue
defineProps({foo: {default: () => undefined}});
```

should be used.

**Comment by jacekkarczmarczyk** [comment]
Consider using https://eslint.org/docs/latest/rules/no-empty-function
